### ISSUE TITLE
Hibernate 5.2 method signature incompatibilty

### ISSUE THREAD

**Issue by johnjaylward** [querydsl/querydsl#1917]
I tried out the latest Query DSL 4.1.2 and Hibernate 5.2, and hibernate appears to have changed method signatures.

```
java.lang.NoSuchMethodError: org.hibernate.Session.createQuery(Ljava/lang/String;)Lorg/hibernate/Query;
    at com.querydsl.jpa.hibernate.DefaultSessionHolder.createQuery(DefaultSessionHolder.java:36)
    at com.querydsl.jpa.hibernate.AbstractHibernateQuery.createQuery(AbstractHibernateQuery.java:104)
    at com.querydsl.jpa.hibernate.AbstractHibernateQuery.createQuery(AbstractHibernateQuery.java:97)
    at com.querydsl.jpa.hibernate.AbstractHibernateQuery.fetch(AbstractHibernateQuery.java:174)
```

The `Query` class moved from `org.hibernate` to `org.hibernate.query` so the APIs don't match anymore.

I'm sure there are more broken areas as well.

**Comment by puntogil** [comment]
hi
should be upgrade also geolatte-geom to 1.x series ... (used by hibernate-spatial module)
any plans?
thanks in advance

**Comment by aeisele** [comment]
Any ETA on when QueryDSL would support Hibernate >= 5.2?

**Comment by Shredder121** [comment]
Not yet, we will have to see how we can support multiple Hibernate 5 versions, or if we just only support the latest.

**Comment by vinctzhang** [comment]
So, anyone can provide a workaround for this? thanks.

**Comment by Shredder121** [comment]
You can't work around a class move.
The only thing you can do is not upgrade for the moment.
It's very surprising that Hibernate even _thought_ about doing this in a minor release.

**Comment by kushieda-minori** [comment]
You could create a compatibility layer that uses the old package names and sub-classes the new package name classes. But that causes it's own issues.

**Comment by eleumik** [comment]
Hi, I am not involved in querydsl but I have same issue in some code around hibernate and googling I came here. I would like to add that the new org.hibernate.query.Query extends the old now deprecated org.hibernate.Query so, I think, is not a class move but a change of return type to an extension interface. 

I have this same issue but only with code compiled with hibernate < 5.2. If instead I compile the same code with 5.2 it works.. like source compatibility is preserved but non binary. I now tested and looks like that changing a return type to an interface that is an extension of previous return type indeed breaks binary compatibility.

Having only source compatibility preserved may be OK for an application that uses hibernate directly: upgrade, compile, done.. but for a compiled library that uses hibernate is a pain.. I think one needs two compiled versions of same code, one for <5.2 e one for 5.2

**Comment by mrfint** [comment]
+1

**Comment by DivideMeByZero** [comment]
Hi. 
Have you already an idea when we can count with the solution?

**Comment by Shredder121** [comment]
Not yet, but I'll focus on it more.

**Comment by Kernle32DLL** [comment]
+1

How about an querydsl 5, targeting hibernate 5.2? As a trade off maybe stop support for querydsl 3 (after a final service release)?

**Comment by Kernle32DLL** [comment]
Hibernate 5.1.5 was released last week. It did mention possibly being the last release of the 5.1 series.

http://in.relation.to/2017/03/14/hibernate-orm-515-final-release/

So any news on the matter here? I would hate to be stuck with hibernate 5.1 because of querydsl.

**Comment by eleumik** [comment]
Hibernate 5.2 should be named Hibernate 6 ;-) I solved this in my project [making an abstract class](https://gist.github.com/eleumik/c25665a785c7d089fb3890daf8b4e4c0) with two source-identical [implementations](https://gist.github.com/eleumik/d64ac9665812cdbe70934a360c5a0bb3), one compiled under 5.1 and one under 5.2, in 2 separate jars.  Then I use an helper  class, [HibernateQuery](https://gist.github.com/eleumik/e40abe01dfc93b231f6ed1803b10842b) compiled under 5.1 to load the right implementation depending on the environment and to expose the implementation statically. Last, all code that use the session to create a query must be changed to call the helper class. You can use the code if you want.

**Comment by Kernle32DLL** [comment]
@eleumik : Food for thought - maybe this could be done via reflection. I do remember thats how Spring Boot implemented compatibility with both Thymeleaf 2 and 3 (which are quite different). Not a very clean solution, but maybe a quick-win to get querydsl running with hibernate 5.2?

https://github.com/spring-projects/spring-boot/blob/1.4.x/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/thymeleaf/ThymeleafAutoConfiguration.java

**Comment by johnjaylward** [comment]
Any word on a resolution for this?

**Comment by mayras** [comment]
Really big issue, I migrated big project and have this issue :(

**Comment by Maze-fr** [comment]
Hibernate has changed a lot about dealing with sessions.
They decided to go closer to JPA specifications with 5.2, and so they now ask to use entity managers instead of sessions.

**Comment by johnjaylward** [comment]
Migrating an entire application to Entity Managers instead of Sessions is no small task. If Hibernate truly didn't want us to use Sessions, they should have bumped the version to 6.0 and deprecated them.

**Comment by johnjaylward** [comment]
Reading the [migration guide](https://github.com/hibernate/hibernate-orm/wiki/Migration-Guide---5.2) it appears that sessions are entity managers in 5.2.

> * org.hibernate.SessionFactory now extends javax.persistence.EntityManagerFactory
> * org.hibernate.Session now extends javax.persistence.EntityManager

**Comment by Maze-fr** [comment]
My first thought was about entity manager, because I was forced to migrate from Hibernate 5.0 to 5.2, but I didn't have any problem.

I use Spring Platform Athens-SR5 (so QueryDSL 4.1.4) + forced version of Hibernate to 5.2 (`<hibernate.version>5.2.10.Final</hibernate.version>` into my POM), if it can be of any help.

**Comment by seiyapl** [comment]
One year after... the issue still exists. Do You have any ETA for the fix?

**Comment by johnjaylward** [comment]
the associated PR appears to fix the issue, but it waiting on merge acceptance from the project maintainers who all appear to be MIA.

**Comment by Shredder121** [comment]
Yeah was going to give them a bit more time before taking matters in my own hands.
Unfortunately Timo is the only one that can properly administer the GitHub organization.

**Comment by Kernle32DLL** [comment]
Just a heads up: Spring Boot 2.0.0 will come with Hibernate 5.2 as a **minimum**, effectively breaking QueryDSL support.

https://github.com/spring-projects/spring-boot/issues/7586

**Comment by johnjaylward** [comment]
@Kernle32DLL that wasn't how I read the ticket you referenced. It sounded like they decided to keep full backwards compatibility to me.

**Comment by johnjaylward** [comment]
Oh, nevermind, I see that the commit that went along with the ticket now which removes support for older versions of hibernate.

**Comment by Shredder121** [comment]
thanks for letting us know!
I didn't yet look at the roadmap for spring boot 2, so thanks for the news.
I currently only can deploy Querydsl 4 artifacts but did hear from Timo, so will resolve the Querydsl 3 release problem.
the fix is in the pipeline at least.

**Comment by lpandzic** [comment]
@Shredder121 can we have an ETA when a fix for this might be available for 4.x?

@others is it possible to use Spring Boot 2.x/Spring Data 2 with Hibernate 5.1.x?

**Comment by Shredder121** [comment]
For 4.x I could deploy a version, but not for 3.x
That's the current situation :/

**Comment by johnjaylward** [comment]
@Shredder121 , at this point, I would think that's acceptable and a notice could maybe be posted in the release notes that 3.x isn't getting further maintenance.

It's been a long time since word has been heard from @timowest, I think we should assume he isn't coming back to maintain his wonderful project.

**Comment by lpandzic** [comment]
Also, Spring Data 2 doesn't support Querydsl 3.

**Comment by FearlessHyena** [comment]
The roadmap for Springboot 2 says that the RC1 will be released very soon (Oct 23)
The actual release is expected to be on or before Dec 4th (in time for the SpringOne conference)
I think we should probably consider QDsl 3.x as EoL or just have it follow the Spring 1.x branch till that ends and move forward with 4.x

**Comment by lpandzic** [comment]
@FearlessHyena, yes, but Spring Data is the one that basically mandates Hibernate version and Spring Data 2 2.0.0.RELEASE is [already out](https://github.com/spring-projects/spring-data-jpa/commit/f98af8a98d5ea76eca7e63f40e29f133d538e2f8) so this should be fixed ASAP.
My advice is querydsl 4.2.0 for version number.

**Comment by lpandzic** [comment]
@Shredder121, do you need any help with this issue? I'd like to help if possible to make the solution happen sooner.

**Comment by Shredder121** [comment]
Basically, a verification that my fix is working would be nice.
Is that something you could do? 👍

**Comment by lpandzic** [comment]
Sure 👍

**Comment by lpandzic** [comment]
@Shredder121 any progress on a fix?

**Comment by Shredder121** [comment]
I mean, there's a fix, but I don't have a way to backport it.
Unless Timo can get back at me and have us work out something that works in the long term.
(Also is the fix working? Did you have time to check?)

**Comment by lpandzic** [comment]
I didn't know fix was available, how can I get it?
It's expected that the fix will be backwards incompatible for Hibernate users, but that is problem of Hibernate, Querydsl should simply keep dependencies up to date. Since breaking change is in Hibernate, user split is on them.

**Comment by kartoffelsup** [comment]
@lpandzic I believe it's right here: https://github.com/querydsl/querydsl/pull/2157

**Comment by lpandzic** [comment]
I've built it locally and it doesn't work.
[Link](https://github.com/infobip/infobip-spring-data-jpa-querydsl/blob/spring-data-upgrade/src/test/java/com/infobip/spring/data/SqlServerQueryDslJpaRepositoryTest.java#L188) to test.
You can try it out, you only need docker and run 
`docker run -e 'ACCEPT_EULA=Y' -e 'SA_PASSWORD=<YourStrong!Passw0rd>' -p 1433:1433 -d microsoft/mssql-server-linux:2017-latest`

**Comment by Shredder121** [comment]
Huh, could you give me output, stacktrace, etc (whatever you can get) of that particular test?

**Comment by lpandzic** [comment]
Source query 
```
select {person.*}
from person
inner join personSettings
on person.id = personSettings.personId
where personSettings.id = ?1
```
doesn't get translated into
```
select person.id as id1_0_0_, person.firstName as firstNam2_0_0_, person.lastName as lastName3_0_0_
from person
inner join personSettings
on person.id = personSettings.personId
where personSettings.id = ?
```
Here's the stacktrace:

```
org.springframework.dao.DataIntegrityViolationException: could not prepare statement; SQL [select {person.*}
from person
inner join personSettings
on person.id = personSettings.personId
where personSettings.id = ?]; nested exception is org.hibernate.exception.DataException: could not prepare statement

	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:263)
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:225)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:500)
	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61)
	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:217)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:153)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:135)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.data.repository.core.support.SurroundingTransactionDetectorMethodInterceptor.invoke(SurroundingTransactionDetectorMethodInterceptor.java:61)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212)
	at com.infobip.spring.data.$Proxy85.jpaSqlQuery(Unknown Source)
	at com.infobip.spring.data.SqlServerQueryDslJpaRepositoryTest.shouldBeAbleToJoin(SqlServerQueryDslJpaRepositoryTest.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.springframework.test.context.junit4.statements.RunBeforeTestExecutionCallbacks.evaluate(RunBeforeTestExecutionCallbacks.java:73)
	at org.springframework.test.context.junit4.statements.RunAfterTestExecutionCallbacks.evaluate(RunAfterTestExecutionCallbacks.java:83)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:251)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:190)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: org.hibernate.exception.DataException: could not prepare statement
	at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:118)
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:42)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:111)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.prepareStatement(StatementPreparerImpl.java:182)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl.prepareQueryStatement(StatementPreparerImpl.java:148)
	at org.hibernate.loader.Loader.prepareQueryStatement(Loader.java:1985)
	at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:1915)
	at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:1893)
	at org.hibernate.loader.Loader.doQuery(Loader.java:938)
	at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:341)
	at org.hibernate.loader.Loader.doList(Loader.java:2692)
	at org.hibernate.loader.Loader.doList(Loader.java:2675)
	at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2507)
	at org.hibernate.loader.Loader.list(Loader.java:2502)
	at org.hibernate.loader.custom.CustomLoader.list(CustomLoader.java:335)
	at org.hibernate.internal.SessionImpl.listCustomQuery(SessionImpl.java:2162)
	at org.hibernate.internal.AbstractSharedSessionContract.list(AbstractSharedSessionContract.java:1016)
	at org.hibernate.query.internal.NativeQueryImpl.doList(NativeQueryImpl.java:152)
	at org.hibernate.query.internal.AbstractProducedQuery.list(AbstractProducedQuery.java:1414)
	at org.hibernate.query.Query.getResultList(Query.java:146)
	at com.querydsl.jpa.sql.AbstractJPASQLQuery.getResultList(AbstractJPASQLQuery.java:206)
	at com.querydsl.jpa.sql.AbstractJPASQLQuery.fetch(AbstractJPASQLQuery.java:238)
	at com.infobip.spring.data.SqlServerQueryDslJpaRepositoryTest.lambda$shouldBeAbleToJoin$6(SqlServerQueryDslJpaRepositoryTest.java:205)
	at com.infobip.spring.data.SimpleExtendedQueryDslJpaRepository.jpaSqlQuery(SimpleExtendedQueryDslJpaRepository.java:80)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.data.repository.core.support.RepositoryComposition$RepositoryFragments.invoke(RepositoryComposition.java:377)
	at org.springframework.data.repository.core.support.RepositoryComposition.invoke(RepositoryComposition.java:200)
	at org.springframework.data.repository.core.support.RepositoryFactorySupport$ImplementationMethodExecutionInterceptor.invoke(RepositoryFactorySupport.java:610)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.doInvoke(RepositoryFactorySupport.java:573)
	at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.invoke(RepositoryFactorySupport.java:554)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.data.projection.DefaultMethodInvokingMethodInterceptor.invoke(DefaultMethodInvokingMethodInterceptor.java:59)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:294)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:98)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:139)
	... 41 more
Caused by: java.sql.SQLException: Unrecognized SQL escape 'person' at line position 14.
	at net.sourceforge.jtds.jdbc.SQLParser.escape(SQLParser.java:1041)
	at net.sourceforge.jtds.jdbc.SQLParser.parse(SQLParser.java:1178)
	at net.sourceforge.jtds.jdbc.SQLParser.parse(SQLParser.java:165)
	at net.sourceforge.jtds.jdbc.JtdsPreparedStatement.<init>(JtdsPreparedStatement.java:111)
	at net.sourceforge.jtds.jdbc.JtdsConnection.prepareStatement(JtdsConnection.java:2492)
	at net.sourceforge.jtds.jdbc.JtdsConnection.prepareStatement(JtdsConnection.java:2450)
	at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:317)
	at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$5.doPrepare(StatementPreparerImpl.java:146)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.prepareStatement(StatementPreparerImpl.java:172)
	... 78 more
```

**Comment by Shredder121** [comment]
Huh. I think we're both referring to different issues.
I'm going to rename this issue, and could you add a new issue?
(also I suppose if the query actually ran, the method invocation did occur, so this issue is fixed?)

**Comment by lpandzic** [comment]
This happens when upgrading from Hibernate 5.1.x to 5.2.
I'll try to isolate the cause and report a new issue.

**Comment by Shredder121** [comment]
Yes, please do that! :+1:

**Comment by lpandzic** [comment]
Done, @Shredder121 , can you please take a look at it? I've isolated the cause to hibernate switch (5.1 to 5.2) so it is technically related to this task.

**Comment by FearlessHyena** [comment]
Hello, just wanted to check if there were any updates?

**Comment by lpandzic** [comment]
@timowest, after seeing you're active on several other issues, can you please shed some light on how this issue should/will be resolved? Spring Boot has been delayed to late February, but after it has been release a lot of users will hit this issue (since Spring Boot 2.0 transitively depends on Hibernate 5.2+)?

**Comment by ask4gilles** [comment]
Hi @Shredder121 , any update on querydsl compatibility with hibernate 5.2? We are really stuck for upgrading to spring boot 2.
Thanks!

**Comment by Shredder121** [comment]
Will make a push for Querydsl 5, which will include dependency updates.
So will also open the PR for suggestions on what to change, since I'm not quite sure what Hibernate Validator versions are usable with which Hibernate versions, and someone was voicing concerns with hibernate-spatial dependency versions as well.
Oh, and we'll probably need to check compatibility with newer Guava versions, as well. :/

**Comment by zenbones** [comment]
Any news on a fix? New QueryDsl version?

**Comment by henriklen** [comment]
Hibernate 5.3 was just released and has a completed task to restore binary compatibility with 5.1 https://hibernate.atlassian.net/browse/HHH-12424 Not sure if this solves all or some parts of the problem but might be good to have in mind when developing a solution.

**Comment by Shredder121** [comment]
In that case we don't need to recompile, can you check if the new hibernate version works now?

**Comment by kasim-ba** [comment]
@Shredder121: So have you performed tests with Hibernate 5.3?

**Comment by Shredder121** [comment]
I have not, do you have an example project I can easily check it on? Or can you check?
Sorry this is left open for so long.

**Comment by lpandzic** [comment]
@Shredder121 any update on a fix for this or Hibernate 5.3?

**Comment by latuszek** [comment]
Seeing how slowly this is going I think the best way is to move away from QueryDSL. The issue has 2 years now.

**Comment by Shredder121** [comment]
@latuszek thanks for the kind words in the time where I need it.

**Comment by latuszek** [comment]
@Shredder121 would any kind worlds help?
I'm looking from the practical perspective. I've been using QueryDSL in my projects for some time, but for more than a year it is stopping me from updating hibernate to newest version. The library is good (otherwise I wouldn't use it), but it's not being updated for some time now. You know what it means in programming world. In new projects I'm avoiding this library because of this. There is nothing unkind about this, I'm just talking about the facts.

**Comment by lpandzic** [comment]
I've seen the original comment on this thread and I'm not sure why deletion or sarcasm is needed but I'd really like querydsl to move forward. In order to do this, querydsl should be opened up more for contributors otherwise things that @latuszek are going to happen. In fact they are already happening and while I was a strong proponent of Querydsl over alternatives, I can no longer do so as a ethical software engineer.
@Shredder121  can we discuss about this somewhere (github/google group/gitter)?

**Comment by NaregB** [comment]
@Shredder121 any updates ?

**Comment by zvozin** [comment]
@Shredder121 need a hand moving on up to Hibernate 5.4? 
Being stuck on 5.1.__ is about to become a show stopper.

**Comment by RayN** [comment]
I managed to make queryDsl work with Sprinboot 2 and Hibernate 5.2 using holon plataform.

The code is here: https://github.com/RayN/queryDslHibernate5SpringBoot2

**Comment by johnjaylward** [comment]
does anyone know how to tell which forks are the most active on GitHub? It's looking like the main repo here needs to be forked and worked on separately. I opened this ticket over 3 years ago. Timo has been missing without word for longer than that, and Shredder doesn't appear to have time, and possibly not permissions to even release changes to master.

**Comment by eleumik** [comment]
I have tested that from Hibernate 5.3.2.Final they restored binary compatibility

**Comment by IdoSalomon** [comment]
> I have tested that from Hibernate 5.3.2.Final they restored binary compatibility

That's great. Thank you for testing, it really helps us move forward with the remaining issues. Closing this for now.
